content/en/blog/_posts/2023-09-15-Bootstrap-an-Air-Gapped-Cluster-with-Kubeadm/index.md

```diff
@@ -0,0 +1,665 @@
+docker tag podinfo/podinfo:6.4.0 localhost:5000/podinfo/podinfo:6.4.0
+docker push localhost:5000/podinfo/podinfo:6.4.0
+```
+
+**Ensure the proper KUBECONFIG is set and install the Podinfo helm chart:**
+
+```
+export KUBECONFIG=/etc/kubernetes/admin.conf
+helm install podinfo ./podinfo-6.4.0.tgz --set image.repository=localhost:5000/podinfo/podinfo
+```
+
+Verify that the Podinfo application comes up:
+```bash
+kubectl get pods -n default
+```
+Or run k9s (a terminal user interface for Kubernetes):
+```bash
+k9s
+```
+## Zarf
+
+Zarf is an open-source tool that takes a declarative approach to software packaging and delivery, including air gap. This same podinfo application will be installed onto the air gap VM using Zarf in this section. The first step is to install [Zarf](https://zarf.dev/install/) on the host/laptop.
+
+Alternatively, a prebuilt binary can be downloaded onto the host/laptop from [GitHub](https://github.com/defenseunicorns/zarf/releases/) for various OS/CPU architectures.
+
+A binary is also needed across the air gap on the VM:
+```bash
+UARCH=$(uname -m)
+# Set the architecture variables if needed
+if [["$UARCH" == "arm64" || "$UARCH" == "aarch64"]]; then
+
+    ARCH="aarch64"
+    K8s_ARCH="arm64"
+
+else
+
+    ARCH="x86_64"
+    K8s_ARCH="amd64"
+
+fi
+
+export ZARF_VERSION=v0.28.3
+
+curl -LO "https://github.com/defenseunicorns/zarf/releases/download/${ZARF_VERSION}/zarf_${ZARF_VERSION}_Linux_${K8s_ARCH}"
+```
+Zarf needs to bootstrap itself into a Kubernetes cluster through the use of an init package. That also needs to be transported across the air gap so let's download it onto the host/laptop:
+```bash 
+curl -LO "https://github.com/defenseunicorns/zarf/releases/download/${ZARF_VERSION}/zarf-init-${K8s_ARCH}-${ZARF_VERSION}.tar.zst"
+```
+The way that Zarf is declarative is through the use of a zarf.yaml file. Here is the zarf.yaml file that will be used for this Podinfo installation. Write it to whatever directory you you have write access to on your host/laptop; your home directory is fine:
+```
+echo "kind: ZarfPackageConfig
+metadata:
+    name: podinfo
+    description: "Deploy helm chart for the podinfo application in K8s via zarf"
+components:
+    - name: podinfo
+        required: true
+        charts:
+            - name: podinfo
+              version: 6.4.0
+              namespace: podinfo-helm-namespace
+              releaseName: podinfo
+              url: https://stefanprodan.github.io/podinfo
+        images:
+        - ghcr.io/stefanprodan/podinfo:6.4.0" > zarf.yaml
+```
+The next step is to build the Podinfo package. This must be done from the same directory location where the zarf.yaml file is located.
+```bash
+zarf package create --confirm
+```
+This command will download the defined helm chart and image and put them into a single file written to disk. This single file is all that needs to be carried across the air gap:
+
+```bash
+$ ls zarf-package-*
+
+zarf-package-podinfo-arm64.tar.zst
```

Comment:
Not a valid shell script - remove the prompt and put the sample output in its own block?

Reply:
👍 , done.